tt_metal/impl/device/device_mesh.hpp

```diff
@@ -54,8 +54,8 @@ class DeviceMesh
     tt::ARCH arch() const;
 
     void close_devices();
-    const DeviceMeshView* get_view() const;
-    DeviceMeshView* get_view();
+    const std::shared_ptr<DeviceMeshView> get_view() const;
```

Comment:
I think you can return reference to shared pointer and who needs a copy can copy it.

Reply:
I'm not familiar with all the details, but from my understanding of the problem, this will open us up to the same issue that this PR is aiming to resolve. I prefer we go default safe and opt-in to something more performant if the more performant option is only conditionally safe.

Again, I'm not fully clear on object lifetimes here but I think we can only return a reference we we know the lifetime of the callee object is strictly longer than the caller scope and I think in that case we would just return a non-owning pointer to indicate that. (for what it's worth, I don't believe the current design in any way makes that guarantee).

Reply:
Yes, I agree with @SeanNijjar here. I intentionally opted for a default safe behavior by doing return by value so that it will avoid a hazard on the object lifetime.